code/modules/urist/modules/GOAI/helpers.dm

```diff
@@ -63,3 +63,36 @@
 
 	return result
 
+# ifdef GOAI_SS13_SUPPORT
+
+/proc/resolve_weakref(var/weakref/wref)
+	if(!wref)
+		return
+	if(istype(wref))
+		return wref.resolve()
+	else
+		return wref
```

Comment:
This should probably take in a generic var, type-cast and istype inside itself. This way the else-clause makes sense; I think it will currently return null if we pass in a non-weakref.

Reply:
I don't _think_ it'll return null as putting the type as a parameter isn't strict/type checked; it can still be a type other than weakref here. LN69 should only return null if the wref argument is null, with LN71 doing the actual type checking.

I can verify this behaviour if you'd like, but I'm pretty sure that's how DM handles it?

Reply:
Really? God, I hate this language so much, it's not even _consistent_ in its jank.

Don't worry about these comments too much; I mainly left them for _myself_ so that I can refer back to them when I'm making adjustments post-merge - though ofc if you know some quirk of DM/some design here and have the time, I appreciate the feedback!